### ISSUE TITLE
incorrect control flow in client.c

### ISSUE THREAD

**Issue by darix** [brianmario/mysql2#1062]
```
client.c: In function 'rb_set_ssl_mode_option':
client.c:140:1: error: control reaches end of non-void function [-Werror=return-type]
  140 | }
      | ^
```

**Comment by darix** [comment]
for completeness. the preprocessed version of the client.c file:

```
static VALUE rb_set_ssl_mode_option(VALUE self, VALUE setting) {
  unsigned long version = mysql_get_client_version();

  if (version < 50703) {
    rb_warn( "Your mysql client library does not support setting ssl_mode; full support comes with 5.7.11." );
    return ((VALUE)RUBY_Qnil);
  }

  mysql_client_wrapper *wrapper; ((wrapper) = (mysql_client_wrapper*)rb_data_object_get(self));;
  int val = rb_num2int_inline(setting);
  if (version >= 50703 && version < 50711) {
    if (val == 1 || val == 3) {
      my_bool b = ( val == 3 );
      int result = mysql_options( wrapper->client, MYSQL_OPT_SSL_ENFORCE, &b );
      return (((VALUE)((int)(result)))<<1 | RUBY_FIXNUM_FLAG);
    } else {
      rb_warn( "MySQL client libraries between 5.7.3 and 5.7.10 only support SSL_MODE_DISABLED and SSL_MODE_REQUIRED" );
      return ((VALUE)RUBY_Qnil);
    }
  }
# 140 "client.c"
}
```

**Comment by darix** [comment]
ok it is actually worse:

Apply this little patch

```
Index: a/ext/mysql2/client.c
===================================================================
--- a/ext/mysql2.orig/client.c
+++ b/ext/mysql2/client.c
@@ -122,6 +122,10 @@ static VALUE rb_set_ssl_mode_option(VALU
       return Qnil;
     }
   }
+  else {
+      rb_warn( "It seems you tried to set ssl options but this library does not know how to do that. Ooops" );
+      return Qnil;
+  }
 #endif
 #ifdef FULL_SSL_MODE_SUPPORT
   GET_CLIENT(self);

```

And you get ...  so we silently failed to set ssl options since a while

```
[8] pry(main)> client = Mysql2::Client.new(:host => "localhost", :username => "root", :ssl_mode => 'SSL_MODE_REQUIRED')
/usr/lib64/ruby/gems/2.6.0/gems/mysql2-0.5.2/lib/mysql2/client.rb:51: warning: It seems you tried to set ssl options but this library does not know how to do that. Ooops
```

**Comment by sodabrew** [comment]
I'm thinking this warning text:
> "mysql2 SSL error: your mysql2 gem was compiled with a mysql client library older than 5.7.11, but your runtime library is newer."

**Comment by darix** [comment]
that is unrelated to this bug.

**Comment by sodabrew** [comment]
Could you confirm my PR 1088 is what you're expecting?

**Comment by darix** [comment]
well at least we no longer have undefined code branches and warn users when they try to set ssl_mode but it is ignored. but a better solution might have been that we fix actually setting ssl_mode for other releases.

In any case ... do we have an ETA when this current fix is in a release?

**Comment by sodabrew** [comment]
Fix is released in 0.5.3 posted a few hours ago.
